crates/flipperzero/src/furi/thread.rs

```diff
@@ -13,3 +136,132 @@ pub fn sleep(duration: core::time::Duration) {
         }
     }
 }
+
+/// A unique identifier for a running thread.
+#[cfg(feature = "alloc")]
+pub struct ThreadId(sys::FuriThreadId);
+
+/// A handle to a thread.
+#[cfg(feature = "alloc")]
+pub struct Thread {
+    /// Guaranteed to be UTF-8.
+    name: Option<CString>,
```

Comment:
I originally considered doing this, and then switched to the same approach that `std::thread::Thread` uses internally, so that the names were guaranteed to be UTF-8 (thus enabling us to have `Thread::name() -> Option<&str>` cheaply). This will always be the case for names of threads created in Rust, but isn't guaranteed to be the case for `sys::furi_thread_get_current()`.

We could instead use the name stored in the `FuriThread`, but we'd need to do one of:
- Return `&CStr` instead of `&str`.
- Check on every call that it is valid UTF-8, even if we know that it is because we created it.
- Cache some `bool` inside the `Thread` object, so we only pay the cost of checking for valid UTF-8 on first usage, and then leverage the faster path on subsequent calls.

Reply:
Ah. OK. That makes sense why `std::thread::Thread` stores its own copy. It can't trust that the OS hasn't tainted it with non-UTF-8 values, even if it was the one who set it when the thread was created.

OK. I understand the current behaviour now. Thanks.